Marlin/src/gcode/parser.h

```diff
@@ -260,12 +256,29 @@ class GCodeParser {
   // The value as a string
   static inline char* value_string() { return value_ptr; }
 
-  // Code value as float
-  static inline float value_float() { return value_ptr ? parse_float(value_ptr) : 0.0; }
+  // Float removes 'E' to prevent scientific notation interpretation
+  static inline float value_float() {
+    if (value_ptr) {
+      char *e = value_ptr;
+      for (;;) {
+        const char c = *e;
+        if (c == '\0' || c == ' ') break;
+        if (c == 'E' || c == 'e') {
+          *e = '\0';
+          const float ret = strtof(value_ptr, nullptr);
```

Comment:
Turns out that `strtof` interprets a string starting with 0X as a hexadecimal value. It's a bug, in my opinion. But, maybe there a `float` value like 0x1E3.4A6 is perfectly legal in C. Patching now….